jax/_src/nn/functions.py

```diff
@@ -588,6 +593,7 @@ def softmax(x: ArrayLike,
   See also:
     :func:`log_softmax`
   """
+  numpy_util.check_arraylike("softmax", x, emit_warning=True)
```

Comment:
We can remove `emit_warning` here, becuase `x` is currently validated when passed to `max`.